python/lsst/pipe/base/pipeTools.py

```diff
@@ -44,13 +48,31 @@
 if TYPE_CHECKING:
     from .taskFactory import TaskFactory
 
+# TODO: remove this module on DM-40443.
+warnings.warn(
```

Comment:
Doesn't look like anything in lsst_distrib or ci is using this code. GitHub shows no-one using it either (other than forks of pipe_base) so removing this file completely seems entirely possible.

Reply:
Given that I've already got the deprecation in, I figure it'll still be easier for me to leave it as is and drop it on DM-40443.